### PR TITLE
Update and expand README.

### LINKED ISSUE #6: Improve README instructions
The instructions to set up the spack source mirror are not correct. When creating or updating a mirror for an environment, one needs to be sure the environment has been concretized beforehand.

### PR DESCRIPTION
**Pull request by micaeljtoliveira**
Closes #6

### CODE REVIEW

**Comment by micaeljtoliveira** [comment]
@aekiss Would you mind having a look at this whenever you have some time?

File: README.md [hunk 1]
```diff
@@ -71,24 +107,150 @@ and `scratch/ik11` to the storage options.
...
-installed under a subdirectory of `/g/data/ik11/spack/0.20.1/modules`. The
-actual subdirectories depend on the selected environement.
+`/g/data/ik11/spack/0.21.2/opt/` and the corresponding environment modules are
+installed under a subdirectory of `/g/data/ik11/spack/0.21.2/modules`. The
+actual subdirectories depend on the selected environment.
+
+
+## Updating to a new spack version
+
+The way how this repository is set up assumes that no update of spack will take
```
Comment (aekiss) [inline README.md]:
```suggestion
The way this repository is set up assumes that no update of spack will take
```

File: README.md [hunk 2]
```diff
@@ -71,24 +107,150 @@ and `scratch/ik11` to the storage options.
...
+
+`opt/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash:7}`
+
+By adding the architecture, compiler name, and compiler version to the path,
+installations of packages built with different compilers on different
+architectures are allowed to coexist without any clashes. The inclusion of part
+of the spack hash also allows for the same version of a given package to be
+compiled with different options. Note that there is nothing specific to the
+spack environments, as this is not necessary (two environments that require
+exactly the same package build in exactly the same way will share the
```
Comment (aekiss) [inline README.md]:
```suggestion
exactly the same package built in exactly the same way will share the
```

**Review by aekiss — APPROVED** [review]
Thanks @micaeljtoliveira, this looks great - I only had tiny suggestions

**Pull request merged**